src/components/Calculator.js

```diff
@@ -2,7 +2,13 @@ import React, { useState } from 'react';
 import './Calculator.css';
 import calculate from '../logic/calculate';
 
-function Calculator() {
+const Calculator = () => (
+  <div className="container">
+    <ChildComponent />
+  </div>
+);
```

Comment:
- [ ] Excellent attempt to implement the previous reviewer's requested change :clap: . However, you are not passing any props to the child component in this manner. To complete this task, I recommend saving the child component in a separate file and importing it into the Calculator component, where you can pass the start through props from the Calculator component to the child component. The calculator component should include the state. Here are some suggestions to assist you. :point_down: 


### Follow these steps: :point_down: 

1. First create a separate file called `ChildComponent.js` and put this :point_down:  there
#### Note: 
We are using `PropTypes` to fix the props validation so you need to run this command `npm i -S prop-types`  to install `prop-types`
```
import React from 'react';
import PropTypes from 'prop-types';

function ChildComponent({ calcData, handleButtonClick }) {
  return (
    <div className="calculator container">
      <input
        className="inputContainer"
        type="text"
        id="myInput"
        name="name"
        pattern="[0-9+\-*/(). ]+"
        value={calcData.next || calcData.total || ''}
        onChange={() => {}}
      />
      <div className="buttonContainer">
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('AC')}
        >
          AC
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('+/-')}
        >
          +/-
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('%')}
        >
          %
        </button>
        <button
          className="mathButton calculatorButton operationButton"
          type="button"
          onClick={() => handleButtonClick('÷')}
        >
          ÷
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('7')}
        >
          7
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('8')}
        >
          8
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('9')}
        >
          9
        </button>
        <button
          className="mathButton calculatorButton operationButton"
          type="button"
          onClick={() => handleButtonClick('x')}
        >
          x
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('4')}
        >
          4
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('5')}
        >
          5
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('6')}
        >
          6
        </button>
        <button
          className="mathButton calculatorButton operationButton"
          type="button"
          onClick={() => handleButtonClick('-')}
        >
          -
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('1')}
        >
          1
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('2')}
        >
          2
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('3')}
        >
          3
        </button>
        <button
          className="mathButton calculatorButton operationButton"
          type="button"
          onClick={() => handleButtonClick('+')}
        >
          +
        </button>
        <button
          className="calculatorButton zero mathButton"
          type="button"
          onClick={() => handleButtonClick('0')}
        >
          0
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('.')}
        >
          .
        </button>
        <button
          className="mathButton calculatorButton equalsButton"
          type="button"
          onClick={() => handleButtonClick('=')}
        >
          =
        </button>
      </div>
    </div>
  );
}

ChildComponent.propTypes = {
  calcData: PropTypes.shape({
    total: PropTypes.string,
    next: PropTypes.string,
    operation: PropTypes.string,
  }).isRequired,
  handleButtonClick: PropTypes.func.isRequired,
};

export default ChildComponent;

```

2. Update the Calculator component this way :point_down: 

```
import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';
import ChildComponent from './ChildComponent';

const Calculator = () => {
  const [calcData, setCalcData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleButtonClick = (buttonName) => {
    const newData = calculate(calcData, buttonName);
    setCalcData(newData);
  };

  return (
    <div className="container">
      <ChildComponent
        calcData={calcData}
        handleButtonClick={handleButtonClick}
      />
    </div>
  );
};

export default Calculator;

```

### I hope this helps 🤗